crates/subspace-farmer-components/src/sector.rs

```diff
@@ -385,21 +385,36 @@ impl SectorContentsMap {
         &self,
         piece_offset: PieceOffset,
     ) -> impl IndexedParallelIterator<Item = Option<(usize, bool)>> + '_ {
+        let piece_offset = usize::from(piece_offset);
```

Comment:
Re: the par_iter() .. IIUC, it fires of a 32K or so parallel operations, even though we probably have a handful of CPUs. I remember the rayon scheduling overhead itself would add up quite a bit 

If possible, can an alternative be tried: the work items are split `N` ways and the batches are passed into par_iter(), where N = num CPUs(or some small number to get a overlap), and see how these compare

Reply:
I this case we're dealing with `IndexedParallelIterator`, which implies random access.

It will take 32k entries and chunk them across CPU threads (~1000 elements for each of 32 threads on my machine) rather than spawning 32k of threads or tasks. So it does exactly what you suggest to do already.

There is non-indexed parallel iterator that can be created with `.par_bridge()`, that one still uses thread pool of the size of number of threads and uses work stealing to process values, which is less efficient than splitting into chunks beforehand and avoid context switching.

Reply:
I see, thanks